docs/deprecation_policy.rst

```diff
@@ -1,3 +1,96 @@
+#############################
+Versioning and Support Policy
+#############################
+
+Qiskit version numbers follow `Sematinc Versioning <https://semver.org/>`__.
+The version number is comprised of 3 primary components, the major, minor, and
+patch versions. For a version number ``X.Y.Z`` where ``X`` is the major version,
+``Y`` is the minor version, and ``Z`` is the patch version.
+
+Breaking API changes are reserved for major version releases. The **minimum**
+period between major version releases is one year. Minor versions will be
+periodically (currently every three months) published for the current major
+version which adds new features and bug fixes. For the most recent minor version
+there will also be new patch versions published as bugs are identified and fixed
+on that release series.
+
+For the purposes of semantic versioning, the Qiskit public API is considered
+any documented module, class, function, or method that is not marked as private
```

Comment:
I should update the deprecation policy section below I think. There was an ambiguity in expectations for some people around public functions that are in weird modules. For example something like `qiskit.transpiler.passes.layout.vf2_utils.score_layout()` which is not documented in the published api docs and I wrote to be internal but under the deprecation policy wording would be viewed as public. My thinking by explicitly saying documented is to tighten the potential surface and make it clear what's expected to be treated as public.

I'll reword this a bit to be more clear, because I think there are some cases where private methods are part of a stable interface. The best example I can think of are things like interface definitions that have abstract private methods that need to be defined.

Reply:
Yeah, there are a couple of cases where `_*` names are public:

- Advanced-usage low-level methods we commit to stability in, for downstream code to use if they guarantee to uphold the class/safety invariants themselves.  The archetypal example of this is `QuantumCircuit._append`, which is _meant_ to be documented, but I think Sphinx might be overlooking it at the moment.
- Abstract methods in base classes that are explicitly meant for subclasses to override.  These are often prefixed with an underscore to show that users aren't meant to call them directly, but the interface classes call them.  Things like `SingletonInstruction._singleton_lookup_key` fall into this category.